Add section about protecting personal or identifying information (#681)
thoughtbot/guides: security/protecting-personal-or-identifying-information.md

# Protecting Personal or Identifying Information

Data privacy and security should be made a priority when developing software in
an effort to protect personal or identifying information.

> Some individuals, due to their backgrounds or circumstances, might be at a
> higher risk of harmful consequences from privacy violations.

Reviewer: ```suggestion
Data privacy and security is important for everyone. It is especially vital for individuals who, due to their backgrounds or circumstances, might be at a higher risk of harmful consequences from privacy violations.
```

Rewording since we all benefit from PII protection

Examples include:

- Survivors of domestic abuse or those who are trying to escape domestic abuse

> - LGBTQIA+ individuals, especially if their housing is dependent on someone who
>   might evict them upon learning this, or if they're in a number of other unsafe
> circumstances

Reviewer: ```suggestion
- LGBTQIA+ individuals who might be in an unsafe circumstance in relation to their identity, including if their housing is dependent on someone who might evict them because of it
```

> - Political dissidents, asylum seekers, targets of government-sanctioned
>   violence.
> - Undocumented immigrants.

Reviewer: ```suggestion
- Political dissidents, asylum seekers, targets of government-sanctioned
  violence
- Undocumented immigrants
```
Removing periods for consistency. Bullet points usually don't end in periods.
